[pt1][quant] Enable UBSAN test for FBGEMM in dynamic quant test

jianyuh opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#25099 [pt1][quant] Enable UBSAN test for FBGEMM in dynamic quant test**

As title says

Differential Revision: [D16986248](https://our.internmc.facebook.com/intern/diff/D16986248/)

jianyuh committed:
[pt1][quant] Enable UBSAN test for FBGEMM in dynamic quant test

As title says

Differential Revision: [D16986248](https://our.internmc.facebook.com/intern/diff/D16986248/)

dskhudia commented on test/test_quantized.py:
```diff
@@ -532,10 +532,9 @@ def equal_ref(X, params, X_scheme, X2, params2, X2_scheme):
-    " Quantized Linear requires FBGEMM. FBGEMM does not play"
-    " well with UBSAN at the moment, so we skip the test if"
-    " we are in a UBSAN environment.",
+    not torch.fbgemm_is_cpu_supported(),
```
Remove TEST_WITH_UBSAN import at the top.

jianyuh replied:
QNNPACK test also used `TEST_WITH_UBSAN` to disable UBSAN test. Let me check if we can also remove that.

supriyar replied:
We don't want to remove it for QNNPACK just yet. Some kernels still fail with ASAN and have to be fixed.

jianyuh replied:
Will keep the `TEST_WITH_UBSAN` for QNNPACK

jianyuh committed:
Update on "[pt1][quant] Enable UBSAN test for FBGEMM in dynamic quant test"


As title says

Differential Revision: [D16986248](https://our.internmc.facebook.com/intern/diff/D16986248/)

jianyuh committed:
Update on "[pt1][quant] Enable UBSAN test for FBGEMM in dynamic quant test"


As title says

Differential Revision: [D16986248](https://our.internmc.facebook.com/intern/diff/D16986248/)
